src/mjolnir/restrictionbuilder.cc

```diff
@@ -271,143 +389,164 @@ void build(const std::string& complex_restriction_from_file,
+                    auto last_tile = tile;
+                    if (last_tile->id() != last_edge_id.Tile_Base()) {
+                      lock.lock();
+                      last_tile = reader.GetGraphTile(last_edge_id);
+                      lock.unlock();
+                    }
+                    auto last_de = last_tile->directededge(last_edge_id);
+                    auto end_node = last_de->endnode();
+                    auto end_node_tile = last_tile;
+                    if (end_node_tile->id() != end_node.Tile_Base()) {
+                      lock.lock();
+                      end_node_tile = reader.GetGraphTile(end_node);
+                      lock.unlock();
+                    }
 
-                  // flip the vias because we walk backwards from the search direction
-                  // using the predecessor edges in thor.
-                  std::reverse(vias.begin(), vias.end());
-
-                  ComplexRestrictionBuilder complex_restriction;
-                  complex_restriction.set_from_id(tmp_ids.at(tmp_ids.size() - 1));
-                  complex_restriction.set_via_list(vias);
-                  complex_restriction.set_to_id(tmp_ids.at(0));
-                  complex_restriction.set_type(restriction.type());
-                  complex_restriction.set_modes(restriction.modes());
-
-                  TimeDomain td = TimeDomain(restriction.time_domain());
-                  if (td.td_value()) {
-                    complex_restriction.set_begin_day_dow(td.begin_day_dow());
-                    complex_restriction.set_begin_hrs(td.begin_hrs());
-                    complex_restriction.set_begin_mins(td.begin_mins());
-                    complex_restriction.set_begin_month(td.begin_month());
-                    complex_restriction.set_begin_week(td.begin_week());
-                    complex_restriction.set_dow(td.dow());
-                    complex_restriction.set_dt(true);
-                    complex_restriction.set_dt_type(td.type());
-                    complex_restriction.set_end_day_dow(td.end_day_dow());
-                    complex_restriction.set_end_hrs(td.end_hrs());
-                    complex_restriction.set_end_mins(td.end_mins());
-                    complex_restriction.set_end_month(td.end_month());
-                    complex_restriction.set_end_week(td.end_week());
-                  }
+                    for (size_t i = 0; i < end_node_tile->node(end_node)->edge_count(); ++i) {
+                      GraphId next_edge_id(end_node_tile->id().tileid(), end_node_tile->id().level(),
+                                           end_node_tile->node(end_node)->edge_index() + i);
+                      auto de = end_node_tile->directededge(next_edge_id);
+                      // reader, lock, from_node, edge_id, from_tile
+                      auto opp_id = GetOpposingEdge(reader, lock, end_node_tile, end_node, de);
+                      if (opp_id != last_edge_id && !de->is_shortcut() &&
```

Comment:
wondering if we should skip platform & egress & transit like we do while matching the osmrestriction https://github.com/valhalla/valhalla/blob/da75fe9b757123faf172511c25b54570f666d09f/src/mjolnir/restrictionbuilder.cc#L67

Reply:
@merkispavel yes we should skip them.  I would just run a quick numbers check(a before and after) on a small state to make sure all is good.

Reply:
ty, done